From 0361594d142e9cec340ef59751132a71c089b1f2 Mon Sep 17 00:00:00 2001
From: Koro <koro@kzxiv.net>
Date: Sun, 9 Aug 2020 07:05:42 -0400
Subject: [PATCH] Fix issues with size_type.

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 xbmc/pictures/PictureInfoTag.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/xbmc/pictures/PictureInfoTag.cpp b/xbmc/pictures/PictureInfoTag.cpp
index d9a6ef76e1..42da3b35ac 100644
--- a/xbmc/pictures/PictureInfoTag.cpp
+++ b/xbmc/pictures/PictureInfoTag.cpp
@@ -176,7 +176,7 @@ void CPictureInfoTag::Archive(CArchive& ar)
     ar << m_exifInfo.LargestExifOffset;
     ar << m_exifInfo.LightSource;
     ar << m_exifInfo.MeteringMode;
-    ar << m_exifInfo.DateTimeOffsets.size();
+    ar << static_cast<int>(m_exifInfo.DateTimeOffsets.size());
     ar << m_exifInfo.Orientation;
     ar << m_exifInfo.Process;
     ar << m_exifInfo.ThumbnailAtEnd;
@@ -303,9 +303,9 @@ void CPictureInfoTag::Serialize(CVariant& value) const
   for (std::vector<int>::size_type i = 0; i < MAX_DATE_COPIES; ++i)
   {
     if (i < m_exifInfo.DateTimeOffsets.size())
-      value["datetimeoffsets"][i] = m_exifInfo.DateTimeOffsets[i];
+      value["datetimeoffsets"][static_cast<int>(i)] = m_exifInfo.DateTimeOffsets[i];
     else
-      value["datetimeoffsets"][i] = static_cast<int>(0);
+      value["datetimeoffsets"][static_cast<int>(i)] = static_cast<int>(0);
   }
   value["digitalzoomratio"] = m_exifInfo.DigitalZoomRatio;
   value["distance"] = m_exifInfo.Distance;
@@ -326,7 +326,7 @@ void CPictureInfoTag::Serialize(CVariant& value) const
   value["largestexifoffset"] = m_exifInfo.LargestExifOffset;
   value["lightsource"] = m_exifInfo.LightSource;
   value["meteringmode"] = m_exifInfo.MeteringMode;
-  value["numdatetimetags"] = m_exifInfo.DateTimeOffsets.size();
+  value["numdatetimetags"] = static_cast<int>(m_exifInfo.DateTimeOffsets.size());
   value["orientation"] = m_exifInfo.Orientation;
   value["process"] = m_exifInfo.Process;
   value["thumbnailatend"] = m_exifInfo.ThumbnailAtEnd;
-- 
2.27.0

